StandardFiniteFieldsOSCAR.jl

```diff
@@ -1,443 +1,443 @@
-    nfactors = sort([r for (r,e) in nfactorization]);
-    lastfactor = nfactors[end]
-    nK = div(n,lastfactor)
-    K = standardfinitefield(p, nK)
-
-    stn = steinitznumberforprimedegree(p, lastfactor, nfactorization[lastfactor])
-    n1 = lastfactor^(nfactorization[lastfactor]-1)
-    q1 = p^n1
-
-    # for each element y in this list, we want to
-    # 1. call EmbedSteinitz(p, n1, nK, y)
-    # 2. this should give a number, we want to use ElementSteinitzNumber to get an element of K.
-    l = digits(stn, base = BigInt(q1))
-    c = map(y -> elementfromsteinitznumber(K, embedsteinitz(p, n1, nK, y)), l)
-    b = elementfromsteinitznumber(K, p^( findfirst(x -> x == div(nK, n1), std_mon_degs(nK))-1))
-
-    L = _extensionwithtowerbasis(K, lastfactor, c, b)
-    ext[n] = L
-    return L
-end
-
-
-
-
-GAP.Packages.load("StandardFF")
-
-function MyPoly(p,n)
-    return defining_polynomial(standardfinitefield(p,n))
-end
-
-function GAPPoly(p,n)
-    poly = GAP.evalstr( "StandardFiniteField(" * string(p) * ", " * string(n) * ")!.DefiningPolynomial!.CoefficientsOfUnivariatePolynomial")
-    poly = polynomial(GF(p), map(x -> GF(p)(x), GAP.gap_to_julia(Vector{GAP.FFE}, poly)))
-end
-
-function compare_poly(p,n)
-  F = GF(p)
-  F.(collect(coefficients(MyPoly(p, n)))) == F.(collect(coefficients(GAPPoly(p, n))))
-end
-
-p = ZZ(3)
-S = Set([ n for n in 6:64 if !compare_poly(p,n) ])
-
-p = ZZ(5)
-for n in 100:150
-    println(n)
-    @time F = standardfinitefield(p,n)
-    @time FG = GAP.Globals.StandardFiniteField(5,n)
-end
+ using Oscar
+
+ # NOTE: These give missing features to OSCAR/Nemo that will likely be added in the near future.
+ function (k::Nemo.FpField)(a::Vector)
+   @assert length(a) == 1
+   return k(a[1])
+ end
+ function (k::FqPolyRepField)(a::Vector)
+   return k(polynomial(GF(ZZ(characteristic(k))), a))
+ end
+ function coords(x::FinFieldElem)
+     return absolute_coordinates(x)
+ end
+ function coords(x::T) where T <: Union{fpFieldElem, FpFieldElem}
+     return [x]
+ end
+
+
+ # NOTE: We piggyback on the _attrs (type Dict{Symbol, Any}) property to define:
+ # is_standardfinitefield (bool)
+ # is_standardprimefield (bool)
+ # primitivepowersintowerbasis (matrix)
+ # towerbasis (also matrix - inverse of primitivepowersintowerbasis)
+ # steinitzprimedegree (double-indexed dict:
+ #                    stpd[r][k] will be integer corresponding to Steinitz number for polynomial fr_k)
+ # standardextensions (Dict{Integer, FField})
+ # NOTE I think a blanket overwrite of getproperty/setproperty! has potential for conflict with other packages
+ #       that also want to use custom properties...
+ # TODO We should give some type checking on these bonus properties
+ # TODO and also redefine propertynames
+ _propertysymbols = [:is_standardfinitefield, :is_standardprimefield, :primitivepowersintowerbasis, :towerbasis, :steinitzprimedegree, :standardextensions]
+ function Base.setproperty!(F::FinField, v::Symbol, c)
```

Comment:
Don't do this: don't overload `setproperty!` / `getproperty` unless you have very, VERY good reason. I see no such reasons here.